### ISSUE TITLE
Details of construction for synthesis routes dataset

### ISSUE THREAD

**Issue by hankook** [binghong-ml/retro_star#2]
I am sincerely thankful for this very interesting work and shared code.

I'm currently trying to construct the synthesis route dataset as you described in the paper, but I failed to obtain 299202 training routes. The following details are obtained from my own construction code:
- I used `raw_train.csv` (from [GLN](https://github.com/Hanjun-Dai/GLN)) for USPTO training dataset, and `origin_dict.csv` (from the shared link) for eMolecules.
- I obtained 153755 starting molecules which are appeared in both eMolecules and USPTO training dataset.
- Starting from them, I obtained 217445 synthesizable molecules including intermediate products, which means I obtained 217445 training routes and some of them are sub-routes (i.e., sub-trees) of others.

So I have some questions for construction:
- How to construct the route dataset in detail? I want to know what I'm missing.
- Do you have any plan to upload the synthesis route dataset or the construction script? It will be very helpful for studying multi-step retrosynthesis and training/testing other datasets.
- eMolecules contains 23M molecules but the paper said it contains 231M molecules. I think this is a typo.

Thanks in advance.

Hankook Lee.

**Comment by binghong-ml** [comment]
Hi Hankook,

We followed the procedures described in the paper to construct the dataset. You are actually using the correct approach but I am not sure whether we start with the same USPTO training split at the beginning. 

You can download our training routes [here](https://www.dropbox.com/s/0nezo0xith8bnml/routes_train.pkl?dl=0).

This is a list of 299202 routes. Each route is a list of reactions as in the following example:
['Cc1ccc(C(=O)N2CCC(c3ccc(-c4cnn(C)c4)cc3)CC2)cc1NC(=O)c1ccc(NC(C)C)nc1>>Cc1ccc(C(=O)N2CCC(c3ccc(-c4cnn(C)c4)cc3)CC2)cc1NC(=O)c1ccc(Cl)nc1.CC(C)N', 'Cc1ccc(C(=O)N2CCC(c3ccc(-c4cnn(C)c4)cc3)CC2)cc1NC(=O)c1ccc(Cl)nc1>>O=C(Cl)c1ccc(Cl)nc1.Cc1ccc(C(=O)N2CCC(c3ccc(-c4cnn(C)c4)cc3)CC2)cc1N', 'Cc1ccc(C(=O)N2CCC(c3ccc(-c4cnn(C)c4)cc3)CC2)cc1N>>Cn1cc(-c2ccc(C3CCNCC3)cc2)cn1.Cc1ccc(C(=O)O)cc1N']

Each reaction is a string of "product >> reactantA.reactantB.reactantC".

The size of eMolecules is 23M. Thanks for spotting the typo!

Best,
Binghong

**Comment by hankook** [comment]
Wow. Thank you for sharing the training routes!

Could you upload other splits (validation/test)? If possible, I want to see the original test routes (I think it may be 10k~20k routes) without cleaning. It will be very useful for demonstrating the generalizability of template-free retro models (e.g., Transformer) with your search algorithm (Retro*) toward unseen reaction templates.

Thanks in advance.

Best,
Hankook Lee

**Comment by binghong-ml** [comment]
Hi Hankook,

Please find the [validation set](https://www.dropbox.com/s/ixw6mupdg0zo27a/routes_val.pkl?dl=0) and [test set](https://www.dropbox.com/s/me9fy4zzujdm6t6/routes_test.pkl?dl=0) in the links. About the test set, I am not entirely sure whether this is the version before cleaning, but you can use it to test your ideas anyways.

Binghong

**Comment by hankook** [comment]
Thanks for sharing, but I currently do not have access to the validation/test sets. Could you give access for downloading the files?

Best,
Hankook Lee

**Comment by binghong-ml** [comment]
Thanks for bringing it to our attention! I've fixed the links. Please see if you can access them.

**Comment by hankook** [comment]
Thanks for sharing the data!
